core/consensus/babe/src/tests.rs

```diff
@@ -151,7 +165,8 @@ impl TestNetFactory for BabeTestNet {
 				inherent_data_providers,
 				config,
 				time_source: Default::default(),
-				transaction_pool : Default::default(),
+				// TODO [slashing] get the transaction pool from somewhere.
+				transaction_pool: Default::default(),
```

Comment:
see -- you have trouble writing the test because you've tried to push too much onto the BABE logic.